dataactbroker/handlers/dashboard_handler.py

```diff
@@ -212,3 +221,121 @@ def historic_dabs_warning_summary(filters):
+    else:
+        return None
+
+
+def historic_dabs_warning_graphs(filters):
+    """ Generate a list of submission graphs appropriate on the filters provided
+
+        Args:
+            filters: dictionary representing the filters provided to the historic dashboard endpoints
+
+        Return:
+            JsonResponse of the submission summaries appropriate on the filters provided
+    """
+    sess = GlobalDB.db().session
+
+    validate_historic_dashboard_filters(filters, graphs=True)
+
+    subs_query = sess.query(
+        Submission.submission_id,
+        (Submission.reporting_fiscal_period / 3).label('quarter'),
+        Submission.reporting_fiscal_year.label('fy'),
+        case([
+            (FREC.frec_code.isnot(None), FREC.frec_code),
+            (CGAC.cgac_code.isnot(None), CGAC.cgac_code)
+        ]).label('agency_code'),
+        case([
+            (FREC.agency_name.isnot(None), FREC.agency_name),
+            (CGAC.agency_name.isnot(None), CGAC.agency_name)
+        ]).label('agency_name')
+    ).join(User, User.user_id == Submission.certifying_user_id).\
+        outerjoin(CGAC, CGAC.cgac_code == Submission.cgac_code).\
+        outerjoin(FREC, FREC.frec_code == Submission.frec_code).\
+        filter(Submission.publish_status_id.in_([PUBLISH_STATUS_DICT['published'], PUBLISH_STATUS_DICT['updated']])).\
+        filter(Submission.d2_submission.is_(False)).order_by(Submission.submission_id)
+
+    subs_query = apply_historic_dabs_filters(sess, subs_query, filters)
+
+    # get the submission metadata
+    sub_metadata = OrderedDict()
+    for query_result in subs_query.all():
+        sub_id = query_result.submission_id
+        sub_metadata[sub_id] = {
+            'submission_id': sub_id,
+            'fy': query_result.fy,
+            'quarter': query_result.quarter,
+            'agency': {
+                'name': query_result.agency_name,
+                'code': query_result.agency_code,
+            },
+            'total_warnings': 0,
+            'warnings': []
+        }
+    sub_ids = list(sub_metadata.keys())
+
+    # build baseline results dict
+    results_data = OrderedDict()
+    resulting_files = filters['files'] or FILE_TYPES
+    for resulting_file in sorted(resulting_files):
+        results_data[resulting_file] = copy.deepcopy(sub_metadata)
+
+    if sub_ids:
+        # get metadata for subs/files
+        error_metadata_query = sess.query(
+            Job.submission_id,
+            CertifiedErrorMetadata.file_type_id,
+            CertifiedErrorMetadata.target_file_type_id,
+            CertifiedErrorMetadata.original_rule_label.label('label'),
+            CertifiedErrorMetadata.occurrences.label('instances')
+        ).join(CertifiedErrorMetadata, CertifiedErrorMetadata.job_id == Job.job_id).\
+            filter(Job.submission_id.in_(sub_ids))
+
+        graph_filters = {'files': filters['files'], 'rules': filters['rules']}
```

Comment:
Just had a thought, since the filters are split up now we don't actually have to remake this filters object, just pass the same one in